### PR TITLE
DMT-42 Refactor and add tooltip ratio

### PR DESCRIPTION
**Pull request by eemilj**
## Related issue
- Closes DMT-42

## Proposed changes
- Add ratio value to tooltip

## Additional information
- Big refactor on percentage calculations, moved functions to only be used once and add percentage as a value to data

### CODE REVIEW

**Review by krasen86 — APPROVED** [review]
Looks good to me :)

File: src/donutState.js [hunk 1]
```diff
@@ -54,22 +54,25 @@ export async function createDonutState(mod) {
...
+            percentage: percentage,
+            absPercentage: Math.abs(percentage),
             mark: (m) => (m ? leaf.mark(m) : leaf.mark()),
             tooltip: () => {
                 /* Adding the display name from the colorAxis and yAxis to the tooltip,
                 to get the corresponding leaf data onto the tooltip. */
                 return colorAxisMeta.parts[0].displayName + ": " + leaf.formattedValue() + "\n" +
-                        yAxis.parts[0].displayName + ": " + sumValue(rows, "Y");
+                        yAxis.parts[0].displayName + ": " + sumValue(rows, "Y") + "\n"
+                        + "Ratio: " + percentage + "%";
```
[the remark this answers is not in the document]
Comment (eemilj) [reply src/donutState.js:75]: Do you mean a different order?

[the remark this answers is not in the document]
Comment (eemilj) [reply src/donutState.js:75]: Yeah sure I'll fix that, there actually looks to be some rounding issues on the tooltip, I'll fix that whilst I am at it

**Review by krasen86 — APPROVED** [review]
Looks nice!

**Pull request merged**